src/java/com/sarality/app/view/list/ListViewItemActionPerformer.java

```diff
@@ -0,0 +1,93 @@
+package com.sarality.app.view.list;
+
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+
+import com.sarality.app.view.action.Action;
+import com.sarality.app.view.action.TriggerType;
+
+/**
+ * Performs the Action when an event on an Item in a ListView is triggered.
+ * 
+ * @author abhideep@ (Abhideep Singh)
+ */
+class ListViewItemActionPerformer implements AdapterView.OnItemClickListener, AdapterView.OnItemLongClickListener,
+    View.OnClickListener, View.OnLongClickListener {
+
+  private final static String LIST_POSITION_TAG = "ListPosition";
```

Comment:
Ideally, we should not be passing in associated data using the views anyways. We should be setting simple flight_id or flight_ride_id as tags on the view, load that in the action and then perform the subsequent action logic. Treat the view like an HTTP page rather than a servicer side object to keep the Viw-Model separation

Having said that, if someone does add a Tag on the View with the associated data, there is nothing in the code that we can do to prevent that. If this is what make life easier, let's go with that for now, but let's give some thought to the problem when we encounter such problems so that we can at least explore finding a solution to these problems.